classes/UpgradeTools/Module/ModuleMigration.php

```diff
@@ -0,0 +1,147 @@
+                    $upgradeFiles[] = ['file' => $file, 'version' => $fileVersion];
+                }
+            }
+        }
+
+        usort($upgradeFiles, function ($a, $b) {
+            return version_compare($a['version'], $b['version']);
+        });
+
+        return array_column($upgradeFiles, 'file');
+    }
+
+    /**
+     * @throws LogicException
+     * @throws UpgradeException
+     */
+    public function runMigration(ModuleMigrationContext $moduleMigrationContext): void
+    {
+        if ($moduleMigrationContext->getMigrationFiles() === null) {
+            throw (new LogicException('Module upgrade files are empty, please run needMigration() first.'));
+        }
+
+        foreach ($moduleMigrationContext->getMigrationFiles() as $index => $migrationFilePath) {
+            $this->logger->notice($this->translator->trans('(%s/%s) Applying migration file %s.', [($index + 1), count($moduleMigrationContext->getMigrationFiles()), basename($migrationFilePath)]));
+
+            $methodName = $this->getUpgradeMethodName($migrationFilePath);
+
+            // check if function already exists to prevent upgrade crash
+            if (function_exists($methodName)) {
+                $moduleMigrationContext->getModuleInstance()->disable();
+                throw (new UpgradeException($this->translator->trans('[WARNING] Method %s already exists. Migration for module %s aborted, you can try again later on the module manager. Module %s disabled.', [$methodName, $moduleMigrationContext->getModuleName(), $moduleMigrationContext->getModuleName()])))->setSeverity(UpgradeException::SEVERITY_WARNING);
+            }
+
+            include $migrationFilePath;
+
+            // @phpstan-ignore booleanNot.alwaysTrue (we ignore this error because we load a file with methods)
+            if (!function_exists($methodName)) {
+                $moduleMigrationContext->getModuleInstance()->disable();
+                throw (new UpgradeException($this->translator->trans('[WARNING] Method %s does not exist. Module %s disabled.', [$methodName, $moduleMigrationContext->getModuleName()])))->setSeverity(UpgradeException::SEVERITY_WARNING);
+            }
+
+            // @phpstan-ignore deadCode.unreachable (we ignore this error because the previous if can be true or false)
+            try {
+                if (!$methodName($moduleMigrationContext->getModuleInstance())) {
+                    throw (new UpgradeException($this->translator->trans('[WARNING] Migration failed while running the file %s. Module %s disabled.', [basename($migrationFilePath), $moduleMigrationContext->getModuleName()])))->setSeverity(UpgradeException::SEVERITY_WARNING);
+                }
+            } catch (UpgradeException $e) {
+                $moduleMigrationContext->getModuleInstance()->disable();
+                throw $e;
```

Comment:
We use to have this specific catch block to disable the module with UpgradeExceptions.

If you disable in all cases, you can remove it.

```suggestion
```